src/components/history/History.tsx

```diff
@@ -205,6 +212,27 @@ export default function History(props: HistoryUrlParams) {
       .join(tokenSeparator);
   };
 
+  React.useEffect(() => {
+    async function flaggedLabwareFromHistory(history: HistoryTableEntry[]) {
+      try {
+        const barcodeSet = history.reduce((acc, entry) => {
+          acc.add(entry.sourceBarcode);
+          acc.add(entry.destinationBarcode);
+          return acc;
+        }, new Set<string>());
+        const labwarePromises = Array.from(barcodeSet).map((barcode) => stanCore.FindFlaggedLabware({ barcode }));
+        const labwares = await Promise.all(labwarePromises);
+        const labwareFlagged = labwares
+          .filter((labware) => labware.labwareFlagged.flagged)
+          .map((labware) => labware.labwareFlagged);
+        setFlaggedLabware(labwareFlagged);
+      } catch (error) {
+        setFlaggedLabware([]);
+      }
+    }
+    flaggedLabwareFromHistory(history);
+  }, [history, setFlaggedLabware]);
+
```

Comment:
I am not sure, but I thought we can check if the labware is flagged from the history search result already, no need to send another request to the core, like checking by event type (if contains 'Flag labware')

Reply:
I don't think it is possible l by looking for "Flag labware" events in the history, because the labware might inherit a flag even if the flag event isn't included in the history you're looking at. David will be adding this information in the history query , so the client can be refactored after this.